File: cuidador-service/PetCare.Cuidador/Models/Cuidadores/Cuidador.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PetCareServicios.Models.Cuidadores
{
    public class Cuidador
    {
        [Key]
        public int CuidadorID { get; set; }

        [Required]
        public int UsuarioID { get; set; }

        [Required]
        [StringLength(20)]
        public string DocumentoIdentidad { get; set; } = string.Empty;

        [StringLength(15)]
        public string TelefonoEmergencia { get; set; } = string.Empty;

        [Column(TypeName = "TEXT")]
        public string? Biografia { get; set; }

        [Column(TypeName = "TEXT")]
        public string? Experiencia { get; set; }

        [StringLength(100)]
        public string? HorarioAtencion { get; set; }

        [Column(TypeName = "DECIMAL(10,2)")]
        public decimal? TarifaPorHora { get; set; }

        [Column(TypeName = "DECIMAL(3,2)")]
        public decimal CalificacionPromedio { get; set; } = 0.0m;

        public bool DocumentoVerificado { get; set; } = false;

        public DateTime? FechaVerificacion { get; set; }

        [StringLength(20)]
        public string Estado { get; set; } = "Activo";

        public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;

        public DateTime? FechaActualizacion { get; set; }

        // Nota: No usamos ForeignKey ni Navigation Property porque estamos usando bases de datos separadas
        // La relación se maneja a nivel de aplicación usando el UsuarioID
    }
} 

File: PetCareSolution-frontend/src/models/Cuidador.js
// models/Cuidador.js
export class CuidadorRequest {
  constructor(
    documentoIdentidad = '',
    telefonoEmergencia = '',
    biografia = '',
    experiencia = '',
    tarifaPorHora = 0,
    horarioAtencion = ''
  ) {
    this.documentoIdentidad = documentoIdentidad;
    this.telefonoEmergencia = telefonoEmergencia;
    this.biografia = biografia;
    this.experiencia = experiencia;
    this.tarifaPorHora = tarifaPorHora;
    this.horarioAtencion = horarioAtencion;
  }
}

// models/Cuidador.js
export class CuidadorResponse {
  constructor(data = {}) {
    this.cuidadorID = data.cuidadorID || 0;
    this.usuarioID = data.usuarioID || 0;
    
    // Información del usuario (ahora vendrá poblada)
    this.nombreUsuario = data.nombreUsuario || '';
    this.emailUsuario = data.emailUsuario || '';
    this.telefonoUsuario = data.telefonoUsuario || '';
    
    // Información específica del cuidador
    this.documentoIdentidad = data.documentoIdentidad || '';
    this.telefonoEmergencia = data.telefonoEmergencia || '';
    this.biografia = data.biografia || '';
    this.experiencia = data.experiencia || '';
    this.horarioAtencion = data.horarioAtencion || '';
    this.tarifaPorHora = data.tarifaPorHora || 0;
    this.calificacionPromedio = data.calificacionPromedio || 0;
    this.documentoVerificado = data.documentoVerificado || false;
    this.fechaVerificacion = data.fechaVerificacion || null;
    this.estado = data.estado || 'Activo';
    this.fechaCreacion = data.fechaCreacion || null;
    this.fechaActualizacion = data.fechaActualizacion || null;
  }

}